### PR TITLE
release: Bump versions to `1.14.0`

### PR DESCRIPTION
**Pull request by marclop**
## Description

Updates the agent version to `1.14.0` for the `7.15.0` stack release.

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -21,13 +21,16 @@ endif::[]
...
-- module/apmhttp: add apmhttp.WithClientSpanType ClientOption to set the span type of http client requests {pull}1106[#(1106)]
+https://github.com/elastic/apm-agent-go/compare/v1.14.0...master[View commits]
 
 [[release-notes-1.x]]
 === Go Agent version 1.x
 
+[[release-notes-1.14.0]]
+==== 1.14.0 - 2021/09/22
+
+- module/apmhttp: add apmhttp.WithClientSpanType ClientOption to set the span type of http client requests {pull}1106[#(1106)]
```
Comment (axw) [inline CHANGELOG.asciidoc:32]:
```suggestion
- module/apmhttp: add apmhttp.WithClientSpanType ClientOption to set the span type of http client requests {pull}1106[#(1106)]
- module/apmazure: introduce instrumentation module for Azure blob, queue, and file storage {pull}1109[#(1109)]
- module/apmfiber: introduce instrumentation module for Fiber web framework {pull}999[#(999)]
```

**Review by axw — APPROVED** [review]
Thanks! We forgot to add a couple of changelog entries - can you please add them while you're there?

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (marclop) [reply CHANGELOG.asciidoc:32]: True, I forgot to check before opening the PR, it seems like we added a couple more things we might want to tell users we did? In particular #1003 seems like a good improvement.

Comment (axw) [reply CHANGELOG.asciidoc:32]: Yes, I think we should add changelog entries for:
 - #1003 
 - #1002 
 - #1006
 - #1104

**Review by axw — APPROVED** [review]
LGTM, thank you!

**Pull request merged**